services/ingestion/subscriber.go

```diff
@@ -252,3 +261,47 @@ func (r *RPCSubscriber) blocksFilter() flow.EventFilter {
 		},
 	}
 }
+
+func (r *RPCSubscriber) fetchBlockEvents(
+	ctx context.Context,
+	blockEvents flow.BlockEvents,
+) models.BlockEvents {
+	blkEvents := flow.BlockEvents{
+		BlockID:        blockEvents.BlockID,
+		Height:         blockEvents.Height,
+		BlockTimestamp: blockEvents.BlockTimestamp,
+	}
+	for _, eventType := range r.blocksFilter().EventTypes {
+		recoveredEvents, err := r.client.GetEventsForHeightRange(
+			ctx,
+			eventType,
+			blockEvents.Height,
+			blockEvents.Height,
+		)
+		if err != nil {
+			return models.NewBlockEventsError(err)
+		}
+
+		if len(recoveredEvents) > 1 {
+			return models.NewBlockEventsError(
+				fmt.Errorf(
+					"received multiple Flow block events for height: %d",
+					blockEvents.Height,
+				),
+			)
+		}
+
+		if len(recoveredEvents) == 0 {
+			return models.NewBlockEventsError(
+				fmt.Errorf(
+					"received empty Flow block events for height: %d",
+					blockEvents.Height,
+				),
+			)
+		}
```

Comment:
```suggestion
		if len(recoveredEvents) != 1 {
		         return models.NewBlockEventsError(
				fmt.Errorf(
					"received %d but expected 1 event for height %d",
					len(recoveredEvents),
					blockEvents.Height,
				),
			)
		}
```

Reply:
Nice ok :ok_hand: 
Updated in https://github.com/onflow/flow-evm-gateway/pull/529/commits/6ff9cb554e41b11d3cc50c3f8890ed3715b77b33